### DISCUSSION TITLE
LTSP problems

### DISCUSSION THREAD

**Discussion by Prof102** [ltsp/ltsp#874] [General] [answered]
Hello everyone,
I've tried to install LTSP, but I'm faced with a series of problems that I can't solve.

1°) The classroom in which I'm trying to install LTPS is made up of 30 clients computers, all of them over 10 years old: Compaq, HP, with no hard disks.
2°) A school router controls I/O on the net.
3°) A computer (Dell Précision) running Linux, with a 500 G° SSD and 16 G° RAM that acts as a server for the room's computers and that I use as my main computer, on which I have two users: a "student" account and a "teacher" account. I have set up a fixed IP address on this computer.
4°) I have followed the instructions on "https://ltsp.org/docs/installation/".
I installed the system Mint 21 with XFCE on the main computer.

After a 1st attempt with Mint 21 which worked for a few hours, I had numerous problems (crashes, inability to load the image on clients, student accounts no longer opening, etc).
I tried reinstalling the system with MINT 21 and then with Debian 12, but in both cases, the installation failed, because I could no longer download "ltsp-binaries" (although I was able to with the 1st attempt with MINT ????). In the end, I reconnected the clients' hard disks and installed grub-ipxe on Mint, but as it still didn't work, I tried Debian 12 + ipxe.

My question: Could you please tell me which Linux system you used to install LTSP and how you set up the installation on the main computer? 

Thanks very much for helping me.

Marc

**Comment by alkisg** [comment, marked as answer]
1) It's fine

2) It's fine

3) It's fine, except that you need one account for EACH student. Not just a single student account.
Also, how do you have a fixed IP if you can't control the router?
E.g. imagine that you set IP=192.168.1.10 in your LTSP server.
Then you power it off and go home.
Next day, suppose that the secretary goes to school before you, and the router gives IP=192.168.1.10 to their PC.
THEN you power on the LTSP server and you try to use your "static" IP=192.168.1.10.
That's an IP conflict and can cause all the issues that you saw.
So, the only correct ways to have a static IP are:
a) To set a "DHCP reservation" on the router, i.e. specify the MAC/IP there, or
b) To set a static IP outside the router range, e.g. if the router gives from 192.168.1.20 to .250, then .10 can be static.

4) To get ltsp-binaries, you need to add the ltsp ppa, whether on Ubuntu, Mint or Debian: 
https://ltsp.org/docs/ppa/

But it's not necessary, it's just "nice to have". LTSP should also work without it.

5) In https://ltsp.org/docs/installation/ I propose Ubuntu MATE LTS as it receives the most testing, and to remove Ubuntu's snap.
But the distribution isn't the issue here.
It sounds like it's a networking issue.
If it's not what I described in (3), then check your cables and also do a LAN benchmark:
https://epoptes.org/documentation/lan-benchmark/

If you don't manage to solve the issue, come to the chat-room for live help:
https://ltsp.org/guides/chat-room/

**Reply by Prof102** [reply]
Hello,

**"It's fine, except that you need one account for EACH student."**
I have over 200 students in total, but no more than 27 students at a time. So I'll create 25 accounts (student1 to student25). But can LTSP handle 25 accounts at once? Isn't there a simpler solution?

**"Also, how do you have a fixed IP if you can't control the router?"**
The high school router IP is set to : 192.168.224.245
My guess, is that this configuration should be ok for my computer :
IP : 192.168.236.77
Subnet mask: 255.255.240.0
Default route : 192.168.224.245

**Epoptes benchmark** look good so far
I'll redo my configuration with these settings and come back to give you feedback.

Thanks for helping.

Marc

**Comment by Prof102** [comment]
Thanks, I'll test your solutions and come back for more information.

Have a nice Sunday
